### ISSUE TITLE
exclude-subviews - not working on 5.4.0 (and :head)

### ISSUE THREAD

**Issue by frapaa** [tombenner/nui#269]
I have an UITextField with a clear-button. The problem is that the border specified in Button in style.nss is also applied to the clear-button inside the text field even though I have set 

<pre>
Button {
    ...
    corner-radius: 0;
    exclude-views: UIAlertButton;
    exclude-subviews: UITableViewCell,UITextField;
}
</pre>
Here's how it looks
![screen shot 2014-08-13 at 16 26 28](https://cloud.githubusercontent.com/assets/1579581/3906792/42815024-22f6-11e4-8e17-4570ce66dd18.png)

Any advice on this?

**Comment by hmlongco** [comment]
Problem is that exclusion class names can have garbage spaces before and after which causes a mismatch. Best to fix it in the parser, but I couldn't track it down. Not sure when this may be applied, but this can also be fixed in UIView+NUI.m. 

- (void)setNuiClass:(NSString*)value
{
    if (![value isEqualToString:kNUIClassNone]) {
        // Set class to none if view is in the exclude
        NSMutableArray *excludeViews = [NSMutableArray arrayWithArray:[[NUISettings get:@"exclude-views" withClass:value] componentsSeparatedByString:@","]];
        // Add global exclusions to the list
        [excludeViews addObjectsFromArray:[NUISettings getGlobalExclusions]];
        if (excludeViews.count) {
            NSString *currentClass = NSStringFromClass([self class]);
            for (NSString *item in excludeViews) {
                if ([item containsString:currentClass]) {
                    value = kNUIClassNone;
                    break;
                }
            }
        }
    }
    
    if (![value isEqualToString:kNUIClassNone]) {
        // Set class to none if any view superviews is in the exclude
        NSMutableArray *excludeSubviews = [NSMutableArray arrayWithArray:[[NUISettings get:@"exclude-subviews" withClass:value] componentsSeparatedByString:@","]];
        // Add global exclusions to the list
        [excludeSubviews addObjectsFromArray:[NUISettings getGlobalExclusions]];
        if (excludeSubviews.count) {
            UIView *superView = self;
            while (superView != nil) {
                NSString *currentClass = NSStringFromClass([superView class]);
                for (NSString *item in excludeSubviews) {
                    if ([item containsString:currentClass]) {
                        value = kNUIClassNone;
                        break;
                    }
                }
                superView = superView.superview;
            }
        }
    }
    
    objc_setAssociatedObject(self, kNUIAssociatedClassKey, value, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
}

**Comment by frapaa** [comment]
So I took a shot at it, and I think I have located the spot where to fix it, but I'm not sure if this will break anything else:
In ```NUIParserDelegate.m``` line 48
change from 
```[contents appendFormat:@" %@", tokenString]; ``` (notice the space before %@)
to
```[contents appendFormat:@"%@", [tokenString stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]]];``` (without the space before %@ and extra measures for other misplaced characters)

**Comment by hmlongco** [comment]
Appreciate it. Looks like you're fixing the root cause of the problem whereas I was only patching the place where the problem manifested itself.

Unfortunately, there are a bunch of pull requests in the system but nothing's been approved in months. I sent an email to Tom asking if he was still working on NUI, but got crickets. 

As such, we decided to go with Pixate instead. Thanks again.

**Comment by SteveNewhouse** [comment]
+1!!!!

**Comment by frapaa** [comment]
Another way around this bug:

```
    UIButton *clearButton = [theTextField valueForKey:@"_clearButton"];
    clearButton.nuiClass = @"none";
```

**Comment by frapaa** [comment]
Yet another way which seems to work better in iOS 9:
```
UIButton *defaultClearButton = [UIButton appearanceWhenContainedIn:[UITextField class], nil];
defaultClearButton.nuiClass = @"none";
```

**Comment by timbodeit** [comment]
Fixed by #312.
